wavedump/src/analyze-waveforms

```diff
@@ -90,134 +83,284 @@ def get_bins(x):
     """
     
     bin_width = 0.5*iqr(x)/(len(x)**(1/3.0))
+    if bin_width == 0:
+        print('Zero bin width! Making it 0.0001.')
+        bin_width = 0.0001
     return np.arange(np.min(x),np.max(x),bin_width)
 
+def chunks(lst, n):
+    """Yield successive n-sized chunks from lst."""
+    for i in range(0, len(lst), n):
+        yield (i,i + n)
+
+def convert_data(f, channel, start, stop):
+    """
+    Reads data from opened hdf5 file `f`.
+    Gets the events from `start` to `stop` in the dataset `channel`.
+    """    
+    if 'data_source' in list(f.attrs):
+        if f.attrs['data_source'] == b'CAEN':
+            
+            xinc = 1/(f.attrs['drs4_frequency'] * 10**6)
+            points = f.attrs['record_length']
+            x = np.linspace(0, xinc * points, int(points)) - xinc * points * (1 - f.attrs['post_trigger']/100)
+
+            y = f[channel][start:stop] * (1/2**12)
+
+    elif 'yinc' in dict(f[channel].attrs):
+        x = f[channel].attrs['xorg'] + np.linspace(0,f[channel].attrs['xinc']*f[channel].attrs['points'],int(f[channel].attrs['points']))
+
+        if True: # ':WAVeform:FORMat' in dict(f['settings'].attrs) and f['settings'].attrs[':WAVeform:FORMat'] != 'ASC':
+            # convert word values -> voltages if the data was saved in a non-ascii format
+            y = f[channel][start:stop]*f[channel].attrs['yinc'] + f[channel].attrs['yorg']
+        else:
+            y = f[channel][start:stop]
+    else:
+        # In older versions of the code, I stored xorg, xinc, etc.
+        # in the main HDF5 group and not on a per channel basis
+        x = f.attrs['xorg'] + np.linspace(0,f.attrs['xinc']*f.attrs['points'],int(f.attrs['points']))
+
+        if ':WAVeform:FORMat' in dict(f['settings'].attrs) and f['settings'].attrs[':WAVeform:FORMat'] != 'ASC':
+            # convert word values -> voltages if the data was saved in a non-ascii format
+            y = f[channel][start:stop]*f.attrs['yinc'] + f.attrs['yorg']
+        else:
+            y = f[channel][start:stop]
+    return x*1e9, y
+
+def low_filter_SPE(time, y):
+    """
+    Returns `y` through a low pass filter. Edit the cutoff frequency by 
+    editing the function itself. `time` is just the temporal spacing between
+    the samples in `y`.
+    """
```

Comment:
I wonder if it's simpler just to pass in x and y?

```python
def low_filter_SPE(x,y):
    dt = x[1] - x[0]
```?

Also, I would change the comment to say "Edit the cutoff frequency by modifying the filter defined below", otherwise it's ambiguous which function you're referring to.